src/debugger/adapter/DebugAdapter.js

```diff
@@ -126,9 +131,23 @@ class PrepackDebugSession extends DebugSession {
         process.exit();
       },
     };
+
     this._adapterChannel.launch(response.request_seq, launchArgs, (dbgResponse: DebuggerResponse) => {
       this.sendResponse(response);
     });
+
+    // The UI will end the configuration sequence by calling 'configurationDone' request.
+    // Sending this during the initializeResponse caused non-determinism in request ordering
+    // that would prevent the debugger from starting.
+    this.sendEvent(new InitializedEvent());
+
+    // It is possible that this gets stuck in an infinte loop that keeps adding the requests
+    // into the arrays if _hasLaunchedDebugServer never becomes true. Because of this, these
+    // loops come after an invarient to ensure the server exists and this situation cannot happen.
+    invariant(this._adapterChannel !== undefined, "Debug Server could not launch");
```

Comment:
We only use invariants to state conditions that we firmly believe should ALWAYS be true. An invariant that could fail because of environmental factors should instead be a runtime check with an appropriate user friendly diagnostic message.